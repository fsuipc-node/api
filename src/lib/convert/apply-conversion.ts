import { VM } from 'vm2';

import { Offset } from '@shared/offset';
import { RawOffsetValue } from '@shared/offset-values';
import { ConvertedOffsetValue } from '@shared/converted-offset-values';

export const applyConversion = (offset: Offset, rawOffsetValue: RawOffsetValue): ConvertedOffsetValue => {
  if (!offset.convert) {
    return rawOffsetValue;
  }

  if (offset.hasMapping) {
    if (offset.isMappingInvalid) {
      return 'INVALID_MAPPING_FUNCTION';
    }

    return offset.mappingFunction(rawOffsetValue);
  } else {
    // tslint:disable-next-line:no-eval
    if (offset.isInvalidConvertExpression) {
      return 'UNSUPPORTED_CONVERSION_EXPRESSION';
    }

    return new VM().run(
      replaceOffsetExpressionValue(offset, rawOffsetValue)
    );
  }
};

export const replaceOffsetExpressionValue = (offset: Offset, rawOffsetValue: RawOffsetValue): string => {
  return offset.convert.replace(
    new RegExp(/{VAL}/g),
    Array.isArray(rawOffsetValue)
      ? `${JSON.stringify(rawOffsetValue)}`
      : typeof rawOffsetValue === 'string'
        ? `'${rawOffsetValue}'`
        : rawOffsetValue.toString()
  );
};
