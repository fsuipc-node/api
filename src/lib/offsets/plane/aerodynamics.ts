import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const APU: OffsetList = {
  incidenceAlpha: new Offset({
    value: 0x2ED0,
    name: 'incidenceAlpha',
    description: 'incidence alpha (body AoA) - radians',
    category: OffsetCategory.AERODYNAMICS,
    type: Type.Double,
    permission: 'r',
  }),
};
