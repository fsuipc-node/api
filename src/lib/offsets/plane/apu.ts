import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const APU: OffsetList = {
  APUGeneratorSwitch: new Offset({
    value: 0xB51,
    name: 'APUGeneratorSwitch',
    category: OffsetCategory.APU,
    description: 'APU generator switch',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  APUGeneratorActive: new Offset({
    value: 0xB52,
    name: 'APUGeneratorActive',
    category: OffsetCategory.APU,
    description: 'APU generator active',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  APUOnFire: new Offset({
    value: 0xB53,
    name: 'APUOnFire',
    category: OffsetCategory.APU,
    description: 'APU generator active',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  APUMaxRPM: new Offset({
    value: 0xB54,
    name: 'APUMaxRPM',
    category: OffsetCategory.APU,
    description: 'APU generator max RPM percent',
    convert: '+({VAL}).toFixed(2)',
    type: Type.Single,
    permission: 'r',
  }),
  APUStarter: new Offset({
    value: 0xB58,
    name: 'APUStarter',
    category: OffsetCategory.APU,
    description: 'APU starter',
    convert: '{VAL} ? 1 : 0',
    type: Type.Single,
    permission: 'rw',
  }),
  APUVoltage: new Offset({
    value: 0xB5C,
    name: 'APUVoltage',
    category: OffsetCategory.APU,
    description: 'APU generator voltage',
    convert: '+({VAL}).toFixed(2)',
    type: Type.Single,
    permission: 'rw',
  }),
};
