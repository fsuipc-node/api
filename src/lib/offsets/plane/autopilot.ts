import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const autopilot: OffsetList = {
  autoPilotAvailable: new Offset({
    value: 0x764,
    name: 'autoPilotAvailable',
    category: OffsetCategory.AUTOPILOT,
    description: 'autopilot available',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  flyByWireELACSwitch: new Offset({
    value: 0x7B6,
    name: 'flyByWireELACSwitch',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire ELAC switch',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  flyByWireELACCompFailFlag: new Offset({
    value: 0x7B7,
    name: 'flyByWireELACCompFailFlag',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire ELAC comp fail flag (read only)',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  flyByWireFACSwitch: new Offset({
    value: 0x7B8,
    name: 'flyByWireFACSwitch',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire FAC switch',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  flyByWireFACCompFailFlag: new Offset({
    value: 0x7B9,
    name: 'flyByWireFACCompFailFlag',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire FAC comp fail flag (read only)',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  flyByWireSECSwitch: new Offset({
    value: 0x7BA,
    name: 'flyByWireSECSwitch',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire SEC switch',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'rw',
  }),
  flyByWireSECCompFailFlag: new Offset({
    value: 0x7BB,
    name: 'flyByWireSECCompFailFlag',
    category: OffsetCategory.AUTOPILOT,
    description: 'fly by wire SEC comp fail flag (read only)',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  apMasterSwitch: new Offset({
    value: 0x7BC,
    name: 'apMasterSwitch',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP master switch',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apWingLevel: new Offset({
    value: 0x7C0,
    name: 'apWingLevel',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP wing level',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apNav1Hold: new Offset({
    value: 0x7C4,
    name: 'apNav1Hold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP NAV1 hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apHeadingHold: new Offset({
    value: 0x7C8,
    name: 'apHeadingHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP heading hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apHeadingValue: new Offset({
    value: 0x7CC,
    name: 'apHeadingValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP heading value - degrees',
    convert: 'Math.round(({VAL} * 360) / 65536)',
    type: Type.UInt16,
    permission: 'rw',
  }),
  apAltitudeHold: new Offset({
    value: 0x7D0,
    name: 'apAltitudeHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP altitude hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apAltitudeValue: new Offset({
    value: 0x7D4,
    name: 'apAltitudeValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP altitude value - ft',
    convert: '+({VAL} * 3.28084 / 65536).toFixed(2)',
    type: Type.Int32,
    permission: 'rw',
  }),
  apAttitudeHold: new Offset({
    value: 0x7D8,
    name: 'apAttitudeHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP attitude hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apAsHold: new Offset({
    value: 0x7DC,
    name: 'apAsHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP airspeed hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apAsValue: new Offset({
    value: 0x7E2,
    name: 'apAsValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP airspeed value - kt',
    type: Type.UInt16,
    permission: 'rw',
  }),
  apMachHold: new Offset({
    value: 0x7E4,
    name: 'apMachHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP mach hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apMachValue: new Offset({
    value: 0x7E8,
    name: 'apMachValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP mach value - mach',
    convert: '{VAL} / 65536',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apVsHold: new Offset({
    value: 0x7EC,
    name: 'apVsHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP vertical speed hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apVsValue: new Offset({
    value: 0x7F2,
    name: 'apVsValue',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP vertical speed value - ft/min',
    type: Type.Int16,
    permission: 'rw',
  }),
  apRPMN1Hold: new Offset({
    value: 0x7F4,
    name: 'apRPMN1Hold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP RPM/N1 hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apRPMN1Value: new Offset({
    value: 0x7FA,
    name: 'apRPMN1Value',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP RPM/N1 value - percent',
    convert: 'Math.round({VAL} * 100 / 16384)',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apGlideSlopeHold: new Offset({
    value: 0x7FC,
    name: 'apGlideSlopeHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP GlideSlope hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apApproachHold: new Offset({
    value: 0x800,
    name: 'apApproachHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP Approach hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apBackCourseHold: new Offset({
    value: 0x804,
    name: 'apBackCourseHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP back course hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apYawDamperHold: new Offset({
    value: 0x808,
    name: 'apYawDamperHold',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP yaw damper hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apTOGAAutoThrottle: new Offset({
    value: 0x80C,
    name: 'apTOGAAutoThrottle',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP TO/GA throttle hold',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apAutoThrottleArm: new Offset({
    value: 0x810,
    name: 'apAutoThrottleArm',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP autothrottle arm',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
  apMaxBank: new Offset({
    value: 0x2E04,
    name: 'apMaxBank',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP max bank - degrees',
    type: Type.UInt32,
    permission: 'r',
  }),
  apMaxSpeed: new Offset({
    value: 0x2F40,
    name: 'apMaxSpeed',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP max speed',
    type: Type.Double,
    permission: 'r',
  }),
  apCruiseSpeed: new Offset({
    value: 0x2F48,
    name: 'apCruiseSpeed',
    category: OffsetCategory.AUTOPILOT,
    description: 'AP cruise speed',
    type: Type.Double,
    permission: 'r',
  }),
};
