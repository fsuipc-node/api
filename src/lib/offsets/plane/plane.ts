import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const plane: OffsetList = {
  vs0: new Offset({
    value: 0x538,
    name: 'vs0',
    category: OffsetCategory.PLANE,
    description: 'Stall speed full flaps - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  vs1: new Offset({
    value: 0x540,
    name: 'vs1',
    category: OffsetCategory.PLANE,
    description: 'Stall speed clean - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  vc: new Offset({
    value: 0x548,
    name: 'vc',
    category: OffsetCategory.PLANE,
    description: 'Cruise speed - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  vmd: new Offset({
    value: 0x550,
    name: 'vmd',
    category: OffsetCategory.PLANE,
    description: 'Minimum drag speed - kt',
    convert: 'ftsecToKt',
    mapping: true,
    type: Type.Double,
    permission: 'r',
  }),
  smokeSystemControl: new Offset({
    value: 0x5D8,
    name: 'smokeSystemControl',
    category: OffsetCategory.PLANE,
    convert: '!!{VAL}',
    description: 'smoke system control on/off',
    type: Type.UInt16,
    permission: 'rw',
  }),
  retractableGear: new Offset({
    value: 0x60C,
    name: 'retractableGear',
    category: OffsetCategory.PLANE,
    description: 'retractable gear (read only)',
    convert: '!!{VAL}',
    type: Type.Byte,
    permission: 'r',
  }),
  retractableLeftFloatExtension: new Offset({
    value: 0x614,
    name: 'retractableLeftFloatExtension',
    category: OffsetCategory.PLANE,
    description: 'retractable left float extension - percent',
    convert: 'Math.floor({VAL} / 16384 * 100)',
    type: Type.UInt16,
    permission: 'r',
  }),
  retractableRightFloatExtension: new Offset({
    value: 0x616,
    name: 'retractableRightFloatExtension',
    category: OffsetCategory.PLANE,
    description: 'retractable right float extension - percent',
    convert: 'Math.floor({VAL} / 16384 * 100)',
    type: Type.UInt16,
    permission: 'r',
  }),
  hasFlaps: new Offset({
    value: 0x778,
    name: 'hasFlaps',
    category: OffsetCategory.PLANE,
    description: 'has flaps',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasStallHorn: new Offset({
    value: 0x77C,
    name: 'hasStallHorn',
    category: OffsetCategory.PLANE,
    description: 'has stall horn',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasMixtureEngine: new Offset({
    value: 0x780,
    name: 'hasMixtureEngine',
    category: OffsetCategory.PLANE,
    description: 'has mixture engine',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasCarbHeat: new Offset({
    value: 0x784,
    name: 'hasCarbHeat',
    category: OffsetCategory.PLANE,
    description: 'has carburator heater',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasSpoilers: new Offset({
    value: 0x78C,
    name: 'hasSpoilers',
    category: OffsetCategory.PLANE,
    description: 'has spoilers',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  isTailDragger: new Offset({
    value: 0x790,
    name: 'isTailDragger',
    category: OffsetCategory.PLANE,
    description: 'has spoilers',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasStrobes: new Offset({
    value: 0x794,
    name: 'hasStrobes',
    category: OffsetCategory.PLANE,
    description: 'has strobes',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasToeBrakes: new Offset({
    value: 0x79C,
    name: 'hasToeBrakes',
    category: OffsetCategory.PLANE,
    description: 'has toe brakes',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  maxMach: new Offset({
    value: 0xB0C,
    name: 'maxMach',
    category: OffsetCategory.PLANE,
    description: 'mach maximum operating - not FS2004',
    convert: '+({VAL} / 20438).toFixed(2)',
    type: Type.Int32,
    permission: 'r',
  }),
  fuelSelectorsCount: new Offset({
    value: 0x123E,
    name: 'fuelSelectorsCount',
    category: OffsetCategory.PLANE,
    description: 'number of fuel selectors for the aircraft',
    type: Type.Byte,
    permission: 'r',
  }),
  aircraftEmptyWeight: new Offset({
    value: 0x1330,
    name: 'aircraftEmptyWeight',
    category: OffsetCategory.PLANE,
    description: 'aicraft empty weight',
    convert: '+({VAL} / 256).toFixed(2)',
    type: Type.Int32,
    permission: 'r',
  }),
  aircraftMaxGrossWeight: new Offset({
    value: 0x1334,
    name: 'aircraftMaxGrossWeight',
    category: OffsetCategory.PLANE,
    description: 'aicraft max gross weight',
    convert: '+({VAL} / 256).toFixed(2)',
    type: Type.Int32,
    permission: 'r',
  }),
};
