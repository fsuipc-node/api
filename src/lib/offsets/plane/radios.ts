import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

export const radios: OffsetList = {
  comFreq: new Offset({
    value: 0x34E,
    name: 'comFreq',
    category: OffsetCategory.RADIOS,
    description: 'Com frequency',
    type: Type.UInt16,
    convert: 'parseInt(`1` + ({VAL}).toString(16))',
    permission: 'rw',
  }),
  transponderFreq: new Offset({
    value: 0x354,
    name: 'transponderFreq',
    category: OffsetCategory.RADIOS,
    description: 'XPND transponder frequency',
    type: Type.UInt16,
    convert: 'parseInt(`1` + ({VAL}).toString(16))',
    permission: 'rw',
  }),
  nav12Select: new Offset({
    value: 0x374,
    name: 'nav12Select',
    category: OffsetCategory.RADIOS,
    description: 'NAV1/NAV2 select',
    type: Type.UInt16,
    permission: 'rw',
  }),
  dme12Select: new Offset({
    value: 0x378,
    name: 'dme12Select',
    category: OffsetCategory.RADIOS,
    description: 'DME1/DME2 select',
    type: Type.UInt16,
    permission: 'rw',
  }),
  navAdfActivate: new Offset({
    value: 0x388,
    name: 'navAdfActivate',
    category: OffsetCategory.RADIOS,
    description: 'NAV and ADF activate < FS2000',
    type: Type.UInt16,
    permission: 'w',
  }),
  comAtisActivate: new Offset({
    value: 0x38A,
    name: 'comAtisActivate',
    category: OffsetCategory.RADIOS,
    description: 'COM/ATIS activate < FS2000',
    type: Type.UInt16,
    permission: 'w',
  }),
  innerMarker: new Offset({
    value: 0xBAC,
    name: 'innerMarker',
    category: OffsetCategory.RADIOS,
    description: 'inner marker - activated when TRUE',
    type: Type.Int16,
    permission: 'r',
  }),
  middleMarker: new Offset({
    value: 0xBAE,
    name: 'middleMarker',
    category: OffsetCategory.RADIOS,
    description: 'middle marker - activated when TRUE',
    type: Type.Int16,
    permission: 'r',
  }),
  outerMarker: new Offset({
    value: 0xBB0,
    name: 'outerMarker',
    category: OffsetCategory.RADIOS,
    description: 'outer marker - activated when TRUE',
    type: Type.Int16,
    permission: 'r',
  }),
};
