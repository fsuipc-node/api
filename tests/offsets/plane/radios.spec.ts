import { radios as offsets } from '@offsets/plane/radios';
import { replaceOffsetExpressionValue } from '@convert/apply-conversion';

describe('offset - plane/radios', () => {
  const offsetsTestCases = [
    { name: 'adf2Freq', value: 74528, expectedResult: 12320 },
    { name: 'adf2RelBearing', value: 38229, expectedResult: 210 },
    { name: 'ndb2IdentSoundSwitch', value: 0, expectedResult: false },
    { name: 'ndb2IdentSoundSwitch', value: 1, expectedResult: true },
    { name: 'vor1DMEDistance', value: 250, expectedResult: 25 },
    { name: 'vor1DMESpeed', value: 350, expectedResult: 35 },
    { name: 'vor1DMETimeToStation', value: 450, expectedResult: 45 },
    { name: 'vor2DMEDistance', value: 550, expectedResult: 55 },
    { name: 'vor2DMESpeed', value: 650, expectedResult: 65 },
    { name: 'vor2DMETimeToStation', value: 750, expectedResult: 75 },
    { name: 'adfFreq', value: 74528, expectedResult: 12320 },
    { name: 'comFreq', value: 13584, expectedResult: 13510 },
    { name: 'nav1Freq', value: 13584, expectedResult: 13510 },
    { name: 'nav2Freq', value: 13584, expectedResult: 13510 },
    { name: 'hasNav1', value: 0, expectedResult: false },
    { name: 'hasNav1', value: 1, expectedResult: true },
    { name: 'hasNav2', value: 0, expectedResult: false },
    { name: 'hasNav2', value: 1, expectedResult: true },
    { name: 'vor2DmeLatitude', value: 5198687, expectedResult: 46.77999650061239 },
    { name: 'vor2DmeLongitude', value: -851715875, expectedResult: -71.38999993912876 },
    { name: 'vor2DmeElevation', value: 92, expectedResult: 301.84 },
    { name: 'vor2ILSLocHeadingTrue', value: 27346, expectedResult: 150.22 },
    { name: 'vor2ILSGlideSlopeAngle', value: 27346, expectedResult: 150.22 },
    { name: 'vor2LocLatitude', value: 5198687, expectedResult: 46.77999650061239 },
    { name: 'vor2LocLongitude', value: -851715875, expectedResult: -71.38999993912876 },
    { name: 'vor2LocElevation', value: 92, expectedResult: 301.84 },
    { name: 'vor2OrILSGlideSlopeLatitude', value: 5198687, expectedResult: 46.77999650061239 },
    { name: 'vor1OrILSGlideSlopeLatitude', value: 5198687, expectedResult: 46.77999650061239 },
    { name: 'vor2OrILSGlideSlopeLongitude', value: -851715875, expectedResult: -71.38999993912876 },
    { name: 'vor1OrILSGlideSlopeLongitude', value: -851715875, expectedResult: -71.38999993912876 },
    { name: 'vor2OrILSGlideSlopeElevation', value: 92, expectedResult: 301.84 },
    { name: 'vor1OrILSGlideSlopeElevation', value: 92, expectedResult: 301.84 },
    { name: 'vor1ILSLocHeadingTrue', value: 27346, expectedResult: 150.22 },
    { name: 'vor1GlideSlopeAngle', value: 27346, expectedResult: 150.22 },
    { name: 'vor1LocLatitude', value: 5198687, expectedResult: 46.77999650061239 },
    { name: 'vor1LocLongitude', value: -851715875, expectedResult: -71.38999993912876 },
    { name: 'vor1LocElevation', value: 92, expectedResult: 301.84 },
    { name: 'vor1DmeLatitude', value: 5198687, expectedResult: 46.77999650061239 },
    { name: 'vor1DmeLongitude', value: -851715875, expectedResult: -71.38999993912876 },
    { name: 'vor1DmeElevation', value: 92, expectedResult: 301.84 },
    { name: 'dme1Distance', value: '92.5', expectedResult: 92.5 },
    { name: 'dme1Distance', value: '92.5 ', expectedResult: 92.5 },
    { name: 'dme1Distance', value: '92.50', expectedResult: 92.5 },
    { name: 'dme1Distance', value: '112.5', expectedResult: 112.5 },
    { name: 'dme1Distance', value: '112.0', expectedResult: 112 },
    { name: 'dme1Distance', value: '112. ', expectedResult: 112 },
    { name: 'dme1Speed', value: '92.5', expectedResult: 92.5 },
    { name: 'dme1Speed', value: '92.5 ', expectedResult: 92.5 },
    { name: 'dme1Speed', value: '92.50', expectedResult: 92.5 },
    { name: 'dme1Speed', value: '112.5', expectedResult: 112.5 },
    { name: 'dme1Speed', value: '112.0', expectedResult: 112 },
    { name: 'dme1Speed', value: '112. ', expectedResult: 112 },
    { name: 'dme2Distance', value: '92.5', expectedResult: 92.5 },
    { name: 'dme2Distance', value: '92.5 ', expectedResult: 92.5 },
    { name: 'dme2Distance', value: '92.50', expectedResult: 92.5 },
    { name: 'dme2Distance', value: '112.5', expectedResult: 112.5 },
    { name: 'dme2Distance', value: '112.0', expectedResult: 112 },
    { name: 'dme2Distance', value: '112. ', expectedResult: 112 },
    { name: 'dme2Speed', value: '92.5', expectedResult: 92.5 },
    { name: 'dme2Speed', value: '92.5 ', expectedResult: 92.5 },
    { name: 'dme2Speed', value: '92.50', expectedResult: 92.5 },
    { name: 'dme2Speed', value: '112.5', expectedResult: 112.5 },
    { name: 'dme2Speed', value: '112.0', expectedResult: 112 },
    { name: 'dme2Speed', value: '112. ', expectedResult: 112 },
    { name: 'nav1MagVar', value: 9102, expectedResult: 50 },
    { name: 'nav2MagVar', value: 9102, expectedResult: 50 },
    { name: 'nav1LocNeedlePosition', value: 127, expectedResult: 100 },
    { name: 'nav1LocNeedlePosition', value: -127, expectedResult: -100 },
    { name: 'nav1GlideslopeNeedlePosition', value: 127, expectedResult: 100 },
    { name: 'nav1GlideslopeNeedlePosition', value: -127, expectedResult: -100 },
    { name: 'nav1GlideslopeActive', value: 0, expectedResult: false },
    { name: 'nav1GlideslopeActive', value: 1, expectedResult: true },
    { name: 'nav1Radial', value: 9102, expectedResult: 50 },
    { name: 'nav2LocNeedlePosition', value: 127, expectedResult: 100 },
    { name: 'nav2LocNeedlePosition', value: -127, expectedResult: -100 },
    { name: 'nav2Radial', value: 9102, expectedResult: 50 },
    { name: 'adf1BearingToNDB', value: 9102, expectedResult: 50 },
    { name: 'nav2GlideslopeNeedlePosition', value: 127, expectedResult: 100 },
    { name: 'nav2GlideslopeNeedlePosition', value: -127, expectedResult: -100 },
    { name: 'nav2GlideslopeActive', value: 0, expectedResult: false },
    { name: 'nav2GlideslopeActive', value: 1, expectedResult: true },
  ];

  describe('offsets list', () => {
    it('should have required properties', () => {
      expect(offsets).toMatchSnapshot();
    });
  });

  offsetsTestCases.forEach(testedOffset => {
    describe(testedOffset.name, () => {
      it('should convert data properly', () => {
        const convertExpression = replaceOffsetExpressionValue(offsets[testedOffset.name], testedOffset.value);

        // tslint:disable-next-line:no-eval
        expect(eval(convertExpression)).toEqual(testedOffset.expectedResult);
      });
    });
  });
});
