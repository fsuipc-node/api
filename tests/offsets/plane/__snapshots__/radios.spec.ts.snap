// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offset - plane/radios offsets list should have required properties 1`] = `
Object {
  "comAtisActivate": Offset {
    "category": "radios",
    "convert": undefined,
    "description": "COM/ATIS activate < FS2000",
    "length": undefined,
    "mapping": undefined,
    "name": "comAtisActivate",
    "permission": "w",
    "type": 5,
    "value": 906,
  },
  "comFreq": Offset {
    "category": "radios",
    "convert": "parseInt(\`1\` + ({VAL}).toString(16))",
    "description": "Com frequency",
    "length": undefined,
    "mapping": undefined,
    "name": "comFreq",
    "permission": "rw",
    "type": 5,
    "value": 846,
  },
  "dme12Select": Offset {
    "category": "radios",
    "convert": undefined,
    "description": "DME1/DME2 select",
    "length": undefined,
    "mapping": undefined,
    "name": "dme12Select",
    "permission": "rw",
    "type": 5,
    "value": 888,
  },
  "innerMarker": Offset {
    "category": "radios",
    "convert": undefined,
    "description": "inner marker - activated when TRUE",
    "length": undefined,
    "mapping": undefined,
    "name": "innerMarker",
    "permission": "r",
    "type": 2,
    "value": 2988,
  },
  "middleMarker": Offset {
    "category": "radios",
    "convert": undefined,
    "description": "middle marker - activated when TRUE",
    "length": undefined,
    "mapping": undefined,
    "name": "middleMarker",
    "permission": "r",
    "type": 2,
    "value": 2990,
  },
  "nav12Select": Offset {
    "category": "radios",
    "convert": undefined,
    "description": "NAV1/NAV2 select",
    "length": undefined,
    "mapping": undefined,
    "name": "nav12Select",
    "permission": "rw",
    "type": 5,
    "value": 884,
  },
  "navAdfActivate": Offset {
    "category": "radios",
    "convert": undefined,
    "description": "NAV and ADF activate < FS2000",
    "length": undefined,
    "mapping": undefined,
    "name": "navAdfActivate",
    "permission": "w",
    "type": 5,
    "value": 904,
  },
  "outerMarker": Offset {
    "category": "radios",
    "convert": undefined,
    "description": "outer marker - activated when TRUE",
    "length": undefined,
    "mapping": undefined,
    "name": "outerMarker",
    "permission": "r",
    "type": 2,
    "value": 2992,
  },
  "transponderFreq": Offset {
    "category": "radios",
    "convert": "parseInt(\`1\` + ({VAL}).toString(16))",
    "description": "XPND transponder frequency",
    "length": undefined,
    "mapping": undefined,
    "name": "transponderFreq",
    "permission": "rw",
    "type": 5,
    "value": 852,
  },
}
`;
